### ISSUE TITLE
`--color=always` doesn't flow through to diffs

### ISSUE THREAD

**Issue by max-sixty** [mitsuhiko/insta#367]
### What happened?

Currently `--color=always` doesn't affect the colorization of diffs 

### Reproduction steps

Take a failing test:

```
cargo insta test --color=always -p prql-compiler --lib --check -- test_upper
```

gives color in the diffs:

<img width="252" alt="image" src="https://user-images.githubusercontent.com/5635139/231007834-9ab31f6d-8279-4331-aa6c-0c64066b611c.png">

But when piped to a file, it only has color in the cargo parts, not the diffs:

<details>

```
[0m[0m[1m[32m    Finished[0m test [unoptimized + debuginfo] target(s) in 0.63s
[0m[0m[1m[32m     Running[0m unittests src/lib.rs (target/debug/deps/prql_compiler-454bbecdaff4c0a9)

running 1 test
test tests::test::test_upper ... FAILED

failures:

---- tests::test::test_upper stdout ----
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ Snapshot Summary ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
Snapshot: upper
Source: prql-compiler/src/tests/test.rs:3436
────────────────────────────────────────────────────────────────────────────────
Expression: compile(
    r#"
    from test_tbles
    derive [upper_name = upper name]
    select [upper_name]
    "#,
)
.unwrap()
────────────────────────────────────────────────────────────────────────────────
-old snapshot
+new results
────────────┬───────────────────────────────────────────────────────────────────
    0     0 │ SELECT␊
    1     1 │   UPPER(name) AS upper_name␊
    2     2 │ FROM␊
    3       │-  test_tables
          3 │+  test_tbles␊
────────────┴───────────────────────────────────────────────────────────────────
Stopped on the first failure. Run `cargo insta test` to run all snapshots.
thread 'tests::test::test_upper' panicked at 'snapshot assertion for 'upper' failed in line 3436', /Users/maximilian/.cargo/registry/src/github.com-1ecc6299db9ec823/insta-1.29.0/src/runtime.rs:569:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    tests::test::test_upper

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 148 filtered out; finished in 0.10s

[0m[0m[1m[31merror[0m[1m:[0m test failed, to rerun pass `-p prql-compiler --lib`
[0m[0m[1m[31merror[0m[1m:[0m 1 target failed:
    `-p prql-compiler --lib`

```

</details>

This also occurs for `CARGO_TERM_COLOR`

### Insta Version

cargo-insta 1.29.0

### rustc Version

1.65

### What did you expect?

As above

**Comment by mitsuhiko** [comment]
Very surprised about this. Both insta (which renders the diff) and cargo-insta use `console` so they should share that setting.

**Comment by xxchan** [comment]
Actually this part by cargo test runner is also not colored.
```
running 1 test
test tests::test::test_upper ... FAILED
```
But by `cargo test --color=always -- --color=always`, according to (https://github.com/rust-lang/cargo/issues/1983), you can have that part colored, but the diff is still not colored. 😄

**Comment by max-sixty** [comment]
If an example is helpful:
- Here's an insta diff: https://github.com/PRQL/prql/actions/runs/5049856404/jobs/9059847037?pr=2601#step:11:225
- Here's a diff from: `pretty_assertions::assert_eq! of almost the same diff: `https://github.com/PRQL/prql/actions/runs/5048986899/jobs/9057951221?pr=2601#step:11:220

**Comment by max-sixty** [comment]
I still don't have a great mental model of this — but at least some of the reason is that `libtest` uses a different option from `cargo`, as @xxchan points out:

> But by `cargo test --color=always -- --color=always`, according to ([rust-lang/cargo#1983](https://github.com/rust-lang/cargo/issues/1983))

...hence the `--color=always -- --color=always`.

The good news is that the outputs might become consolidated under cargo (https://github.com/rust-lang/cargo/issues/1983#issuecomment-1561603937), which might solve then solve this, or at least make it more tractable

**Comment by max-sixty** [comment]
Closed by #375 (I think). Passing `CLICOLOR_FORCE` in CI seems to work; it's still not clean given `--color=` doesn't always work.

**Comment by Sajjon** [comment]
@mitsuhiko I great lib! I'm using `insta`, version = "1.42.1" and I don't see any colors at all, not in VS Code neither in Iterm2. Not when using `cargo test` nor `cargo nextest run` and neither when using `--color=always` nor with `--color=always -- --color=always`. I've also tried exporting variables:
```
CLICOLOR=1
NEXTEST_COLOR=1
CLICOLOR_FORCE=1
```

but to no avail.

I'm on macOS (Apple Silicon M2) Sequioa 15.3.1 using `rustc 1.85.0 (4d91de4e4 2025-02-17)`.

I do however get color highlighting when using `cargo insta test --color=always`.
